### PR TITLE
Issue 357

### PR DESCRIPTION
**Pull request by vincentneko**
This is a big one...
- Move resource count inside cards
- Pass resource count to select card components so that resource count is displayed in all circumstances

### CODE REVIEW

File: src/models/PlayerInputModel.ts [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
+import { ICard } from '../cards/ICard';
 
 export interface PlayerInputModel {
     amount: number | undefined;
     availableSpaces: Array<string> | undefined;
     canUseHeat: boolean | undefined;
     canUseSteel: boolean | undefined;
     canUseTitanium: boolean | undefined;
-    cards: Array<string> | undefined;
+    cards: Array<ICard> | undefined;
```
Comment (bafolts) [inline src/models/PlayerInputModel.ts]: This should be `cards: Array<CardModel> | undefined`. Anything on client side should use the models which attempt a encapsulate data we can serialize to JSON.

File: server.ts [hunk 2]
```diff
@@ -414,15 +414,13 @@ function getWaitingFor(
       }
       break;
     case PlayerInputTypes.SELECT_HOW_TO_PAY_FOR_CARD:
-      result.cards = (waitingFor as SelectHowToPayForCard)
-          .cards.map((card) => card.name);
+      result.cards = (waitingFor as SelectHowToPayForCard).cards;
```
Comment (bafolts) [inline server.ts]: I would create a function `getCards(cards: Array<ICard>): Array<CardModel>` and map these to a `CardModel` for the client.

Comment (vincentneko) [reply server.ts]: > One minor change involving using `CardModel` for the client. Everything else looks great.



> One minor change involving using `CardModel` for the client. Everything else looks great.

I've just found a bug, cards cannot be paid with steel or titanium, I will investigate this.

**Comment by vincentneko** [comment]
> CardType.AUTOMATED

Not sure how to handle this as the input object is a `ICard ` and does not have a `cardType` property...
Any suggestion is welcome

**Comment by vincentneko** [comment]
> > CardType.AUTOMATED
> 
> Not sure how to handle this as the input object is a `ICard ` and does not have a `cardType`
> property...
> The card could also be a corporation.
> Any suggestion is welcome

Sleeping on this one, maybe `cardType `property should be moved to `ICard ` so corporations would also have their `cardType `
I'm merging as it is now, we could make the change with another issue.

**Pull request merged**